test/lib/Elastica/Test/SearchTest.php

```diff
@@ -140,6 +140,14 @@ public function testAddIndexInvalid()
         $search->addIndex(new \stdClass());
     }
 
+    public function testAddNumericIndex()
+    {
+        $client = $this->_getClient();
+        $search = new Search($client);
+
+        $search->addIndex(1);
```

Comment:
I would suggest to also add an assert, that the "index" was successfully added. For example you could use getIndexes() and check if it is part of it.